src/api/publish.rs

```diff
@@ -4,9 +4,16 @@ use uuid::Uuid;
 /// The publish request.
 #[derive(Deserialize, Debug)]
 pub struct PublishRequest {
-    pub name: String,
-    pub version: String,
-    pub upload_id: String,
+    pub package_name: String,
+    pub upload_id: Uuid,
+    pub num: String,
+    pub package_description: Option<String>,
+    pub repository: Option<String>,
+    pub documentation: Option<String>,
+    pub homepage: Option<String>,
+    pub urls: Vec<Option<String>>,
```

Comment:
nit: why not simply `Vec<Option<String>>,`; can we expect `None` elements in the vec?

Reply:
Good question, the reason is that in Postgres, `TEXT[] NOT NULL` still allows nullable values inside the array. There are ways to enforce non-null values such as using a custom domain, but diesel still interprets it as possibly null values inside the array. This type matches what diesel generates.

Reply:
Good point, will update as well as making this type `Url` instead of `String` per Josh's comment.